src/useTheme.test.ts

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only
+Please see LICENSE in the repository root for full details.
+*/
+
+import { renderHook } from "@testing-library/react-hooks";
+import {
+  afterEach,
+  beforeEach,
+  describe,
+  expect,
+  Mock,
+  test,
+  vi,
+} from "vitest";
+
+import { useTheme } from "./useTheme";
+import { useUrlParams } from "./UrlParams";
+
+// Mock the useUrlParams hook
+vi.mock("./UrlParams", () => ({
+  useUrlParams: vi.fn(),
+}));
+
+describe("useTheme", () => {
+  let originalClassList: DOMTokenList;
+  beforeEach(() => {
+    // Save the original classList so we can restore it later
+    originalClassList = document.body.classList;
+
+    vi.spyOn(document.body.classList, "add").mockImplementation(vi.fn());
+    vi.spyOn(document.body.classList, "remove").mockImplementation(vi.fn());
```

Comment:
Generally `.mockImplementation(() => {})` or `.mockImplementation(vi.fn())` is a no-op, so these can likely be

```suggestion
    vi.spyOn(document.body.classList, "add");
    vi.spyOn(document.body.classList, "remove");
```